packages/cli/src/databases/migrations/sqlite/1690000000002-MigrateIntegerKeysToString.ts

```diff
@@ -178,3 +186,44 @@ export class MigrateIntegerKeysToString1690000000002 implements IrreversibleMigr
 		);
 	}
 }
+
+const DESIRED_DATABASE_FILE_SIZE = 1 * 1024 * 1024 * 1024; // 1 GB
+const migrationsPruningEnabled = process.env.MIGRATIONS_PRUNING_ENABLED === 'true';
+
+function getSqliteDbFileSize(): number {
+	const filename = path.resolve(
+		UserSettings.getUserN8nFolderPath(),
+		config.getEnv('database.sqlite.database'),
+	);
+	const { size } = statSync(filename);
+	return size;
+}
+
+const pruneExecutionsData = async ({ queryRunner, tablePrefix }: MigrationContext) => {
+	if (migrationsPruningEnabled) {
+		const dbFileSize = getSqliteDbFileSize();
+		if (dbFileSize < DESIRED_DATABASE_FILE_SIZE) {
+			console.log(`DB Size not large enough to prune: ${dbFileSize}`);
+			return;
+		}
+
+		console.time('pruningData');
```

Comment:
We might want to remove the console.time for production deployment

Reply:
we can do that after the release. I would like to be able to have a bit more visibility for the patch releases.